test/util/validators/FormFieldValidatorSpec.scala

```diff
@@ -179,6 +361,220 @@ class FormFieldValidatorSpec extends WordSpec with MustMatchers {
+        val input = "Special characters a,.-'/"
+
+        isAlphanumericWithAllowedSpecialCharacters(input) must be(true)
+      }
+    }
+  }
+
+  "FormFieldValidator on startsWithCapitalLetter" should {
+    "return false"  when {
+      "input start with lowercase" in {
+        val input = "lowercaseString"
+
+        startsWithCapitalLetter(input) must be(false)
+      }
+
+      "string is empty" in {
+        startsWithCapitalLetter(emptyString) must be(false)
+      }
+    }
+
+    "return true" when {
+      "string start with capital letter" in {
+        val input = "CapitalLetter"
+
+        startsWithCapitalLetter(input) must be(true)
+      }
+    }
+  }
+
+  "FormFieldValidator on isContainedIn" should {
+    "return false" when {
+      "list is empty" in {
+        isContainedIn(List())("element") must be(false)
+      }
+
+      "list doesn't contain specific element" in {
+        isContainedIn(List("A"))("B") must be(false)
+      }
+    }
+
+    "return true" when {
+      "element is on the list" in {
+        isContainedIn(List("A"))("A") must be(true)
+      }
+    }
+  }
+
+  "FormFieldValidator on containsNotOnlyZeros" should {
+    "return false" when {
+      "string contains only zero or zeros" in {
+        containsNotOnlyZeros("0") must be(false)
+        containsNotOnlyZeros("00") must be(false)
+        containsNotOnlyZeros("000") must be(false)
+      }
+    }
+
+    "return true" when {
+      "string contains different digits than 0" in {
+        val input = "1230"
+
+        containsNotOnlyZeros(input) must be(true)
+      }
+
+      "string contains alphabetic characters" in {
+        val input = "asv"
+
+        containsNotOnlyZeros(input) must be(true)
+      }
+
+      "string contains alphabetic characters and 0" in {
+        val input = "0asd00"
+
+        containsNotOnlyZeros(input) must be(true)
+      }
+    }
+  }
+
+  "FormFieldValidator on isTailNumeric" should {
+    "return false" when {
+      "input is empty" in {
+        isTailNumeric("") must be(false)
+      }
+
+      "input contains two or more characters but not only numeric" in {
+        isTailNumeric("12nd12") must be(false)
+      }
+    }
+
+    "return true" when {
+      "input contains only one character" in {
+        isTailNumeric("0") must be(true)
+      }
+
+      "input contains all numeric characters starts from 2nd char" in {
+        isTailNumeric("a1241") must be(true)
+      }
+    }
+  }
+
+  "FormFieldValidator on isDecimal" should {
+    "return false" when {
+      "string contains alphabetic characters" in {
+        isDecimal("123as.213") must be(false)
+      }
+
+      "string is empty" in {
+        isDecimal("") must be(false)
+      }
+
+      "string doesn't contain dot" in {
+        isDecimal("123412") must be(false)
+      }
+    }
+
+    "return true" when {
+      "string contains only digits and dot" in {
+        isDecimal("123456.123456") must be(true)
+      }
+    }
+  }
+
```

Comment:
Added comment on this method in its definition place. We cannot proceed with the current form.